### PR TITLE
upgrade phoenix from 0.13 to 0.15

### PR DESCRIPTION
**Pull request by shankardevy**

### CODE REVIEW

File: web/channels/snapshot_socket.ex [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  #
+  # Would allow you to broadcast a "disconnect" event and terminate
+  # all active sockets and channels for a given user:
+  #
+  #     MyApp.Endpoint.broadcast("users_socket:" <> user.id, "disconnect", %{})
+  #
+  # Returning `nil` makes this socket anonymous.
+  def id(_socket), do: nil
+end
+Leaving
```
Comment (mosic) [inline web/channels/snapshot_socket.ex]: You left "Leaving" here :smile:

**Comment by mosic** [comment]
This change seems to break channels, websockets requests returns 404.

You can test this with http://www.websocket.org/echo.html, try `ws://localhost:4000/ws` from master and your branch.

**Comment by shankardevy** [comment]
@milos I will look into this.

**Comment by shankardevy** [comment]
@mosic would like to test [phoenix_16_upgrade](https://github.com/evercam/evercam-media/tree/phoenix_16_upgrade)? It looks OK for me.

> You can test this with http://www.websocket.org/echo.html, try ws://localhost:4000/ws from master and your branch.

with the latest phoenix 0.16, the url above says `disconnected` even for a  new phoenix project with ws working. So testing with the above link will lead to false positive.

Note: try the `phoenix_16_upgrade` in `evercam-media` and not the one on my fork.

**Comment by mosic** [comment]
@shankardevy I get the same error on that branch too. I think it may be because of the changes to the `router.ex` file.

You can see it for yourself if you run `evercam-media` and `evercam-dashboard` locally, you will get 404 responses for websocket requests on camera single pages.

**Comment by mosic** [comment]
Superseded by https://github.com/evercam/evercam-media/pull/82.

**Pull request closed without merging**